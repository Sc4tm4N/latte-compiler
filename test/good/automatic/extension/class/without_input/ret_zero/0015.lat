class a {

  void print() {
    printString("i am a, my a is:");
    printString(a());
  }

  string a() {
    return "kappa a";
  }
}

class b extends a {

  void print() {
    printString("i am b, my a is:");
    printString(a());
  }

  string a() {
    return b();
  }

  string b() {
    return "kappa b";
  }

  int h() {
    return 3;
  }
}

class c extends b {
  int a;

  int x() {
    return 2;
  }

  void print() {
    printString("i am c, my a is:");
    printString(a());
  }

  string a() {
    return b() + self.c();
  }

  string c() {
    return " huh " + c();
  }
}

string c() {
  return "kappa outside c";
}

int main() {
  a a = new a;
  a.print();

  b b = new b;
  b.print();

  a ab = new b;
  ab.print();

  c c = new c;
  c.print();

  a ac = new c;
  ac.print();

  ac = new b;
  ac.print();

  return 0;
}
