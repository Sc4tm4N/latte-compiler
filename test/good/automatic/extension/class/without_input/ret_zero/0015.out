i am a, my a is:
kappa a
i am b, my a is:
kappa b
i am b, my a is:
kappa b
i am c, my a is:
kappa b huh kappa outside c
i am c, my a is:
kappa b huh kappa outside c
i am b, my a is:
kappa b
